test/cri-containerd/gmsa.go

```diff
@@ -0,0 +1,48 @@
+// +build functional
+
+package cri_containerd
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+// Generates a gmsa credential spec and places it at 'path'. Dir has to exist, it will not be created by
+// New-CredentialSpec.
+func generateCredSpec(path string) error {
+	output, err := exec.Command(
+		"powershell",
+		"New-CredentialSpec",
+		"-AccountName",
+		gmsaAccount,
+		"-Path",
+		path,
+	).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to create new credential spec: %s", string(output))
+	}
+	return nil
+}
+
+// Tries to generate a cred spec to use for gmsa test cases. Returns the cred
+// spec and an error if any.
+func gmsaSetup() (string, error) {
+	csDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		return "", errors.New("failed to create temp directory")
+	}
+	defer os.RemoveAll(csDir)
+	csPath := filepath.Join(csDir, gmsaAccount+".json")
```

Comment:
Rather than creating a temp directory, then picking a file name in that directory and creating it, can we just get a temp file from `ioutil.TempFile` and use that instead?

Reply:
'New-CredentialSpec' is what makes the file with the supplied name. Not sure how that would function

Reply:
If we control the path, and it doesn't enforce any special requirements, we should be able to do this just fine.

Reply:
It creates a new file with the path you provided + a .json extension to the path by default unless one is specified like above. We'd have to create the file with a *.json suffix as the second param, and then close the file immediately (used by another process error) so that New-CredentialSpec can just write over it? Unless I'm missing something here

Reply:
```go
func gmsaSetup(t *testing.T) string {
	f, err := ioutil.TempFile("", "*.json")
	if err != nil {
		t.Fatalf("failed to create temp directory: %s", err)
	}
	name := f.Name()
	f.Close()
	defer os.Remove(name)
	if err := generateCredSpec(name); err != nil {
		t.Fatal(err)
	}
	credSpec, err := ioutil.ReadFile(name)
	if err != nil {
		t.Fatalf("failed to read credential spec: %s", err)
	}
	return string(credSpec)
}
```

Reply:
This look good? @kevpar

Reply:
I would say leave it as is but there's no need to use `gmsaAccount` in the file name, so just change it to `credspec.json` or something.

Reply:
gotcha :)